app/lib/otis/registration_mover.rb

```diff
@@ -0,0 +1,29 @@
+module Otis
+  # Responsible for creating a new HTUser, populating it with the data in
+  # an HTRegistration, and saving.
+  class RegistrationMover
+    def initialize(registration)
+      @registration = registration
+    end
+
+    def ht_user
+      return @ht_user unless @user.nil?
+
+      @ht_user = HTUser.new(userid: @registration.dsp_email,
```

Comment:
The userid won't always be `@registration.dsp_email`; instead, we need to capture that from the `X-Remote-User` header from finalization, and we should capture other headers for display (at least).

Reply:
Let me make sure I get this right, REMOTE_USER becomes ht_user.userid, for the email we can still use otis_registration.dsp_email or even HTTP_X_SHIB_EDUPERSONPRINCIPALNAME. I was thinking I can either store the HTTP request ENV in a JSON structure in a new field, let's call it otis_registration.finalize_env. Are we storing/displaying anything and everything of the form HTTP_X_SHIB_* in the headers? Anything else (beside REMOTE_USER)?

Reply:
`HTTP_X_SHIB_EDUPERSONPRINCIPALNAME` won't reliably be populated or (if it is) reliably an email address or reliably the correct email address, so I would still use `otis_registration.dsp_email`. I don't think we need to store anything from the environment besides `HTTP_X_SHIB_*` and `REMOTE_USER`.

Reply:
It might (?) be preferable to just store the keycard user attributes, as that already takes care of getting them out of the headers and decoding as necessary. OTOH we were thinking about getting rid of keycard entirely so :shrug:

Reply:
:+1: